File: src/main/java/me/aov/sellgui/gui/GUIManager.java
package me.aov.sellgui.gui;

import me.aov.sellgui.SellGUIMain;
import org.bukkit.entity.Player;
import org.bukkit.inventory.Inventory;

import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import java.util.UUID;

public class GUIManager {

    private final SellGUIMain plugin;
    private final Map<UUID, PriceEvaluationGUI> activeEvaluationGUIs = new HashMap<>();
    private final Map<UUID, AutosellSettingsGUI> activeAutosellGUIs = new HashMap<>(); // New map for autosell GUIs

    public GUIManager(SellGUIMain plugin) {
        this.plugin = plugin;
    }

    public void openPriceEvaluationGUI(Player player) {

        if (activeEvaluationGUIs.containsKey(player.getUniqueId())) {
            player.closeInventory();
        }
        PriceEvaluationGUI gui = new PriceEvaluationGUI(plugin, player, plugin.getNBTPriceManager());
        activeEvaluationGUIs.put(player.getUniqueId(), gui);
        player.openInventory(gui.getInventory());
    }

    public PriceEvaluationGUI getActivePriceEvaluationGUI(Player player) {
        return activeEvaluationGUIs.get(player.getUniqueId());
    }

    public PriceEvaluationGUI getActivePriceEvaluationGUI(Inventory inventory) {
        for (PriceEvaluationGUI gui : activeEvaluationGUIs.values()) {
            if (gui.getInventory().equals(inventory)) {
                return gui;
            }
        }
        return null;
    }

    // New method to open the autosell settings GUI
    public void openAutosellSettingsGUI(Player player) {
        if (activeAutosellGUIs.containsKey(player.getUniqueId())) {
            player.closeInventory();
        }
        AutosellSettingsGUI gui = new AutosellSettingsGUI(plugin, player);
        activeAutosellGUIs.put(player.getUniqueId(), gui);
        player.openInventory(gui.getInventory());
    }

    public AutosellSettingsGUI getActiveAutosellSettingsGUI(Player player) {
        return activeAutosellGUIs.get(player.getUniqueId());
    }

    public AutosellSettingsGUI getActiveAutosellSettingsGUI(Inventory inventory) {
        for (AutosellSettingsGUI gui : activeAutosellGUIs.values()) {
            if (gui.getInventory().equals(inventory)) {
                return gui;
            }
        }
        return null;
    }

    public void removePlayer(Player player) {
        activeEvaluationGUIs.remove(player.getUniqueId());
        activeAutosellGUIs.remove(player.getUniqueId()); // Remove from autosell GUIs as well
    }

    public void reload() {

        for (UUID playerUUID : Set.copyOf(activeEvaluationGUIs.keySet())) {
            Player player = plugin.getServer().getPlayer(playerUUID);
            if (player != null && player.getOpenInventory().getTopInventory().getHolder() instanceof PriceEvaluationGUI) {
                player.closeInventory();
            }
        }
        activeEvaluationGUIs.clear();

        // Reload for autosell GUIs
        for (UUID playerUUID : Set.copyOf(activeAutosellGUIs.keySet())) {
            Player player = plugin.getServer().getPlayer(playerUUID);
            if (player != null && player.getOpenInventory().getTopInventory().getHolder() instanceof AutosellSettingsGUI) {
                player.closeInventory();
            }
        }
        activeAutosellGUIs.clear();
    }
}

File: src/main/kotlin/me/aov/sellgui/gui/GUIManager.kt
package me.aov.sellgui.gui

import me.aov.sellgui.SellGUIMain
import org.bukkit.entity.Player
import org.bukkit.inventory.Inventory
import java.util.UUID

class GUIManager(private val plugin: SellGUIMain) {
    private val activeEvaluationGUIs = HashMap<UUID, PriceEvaluationGUI>()
    private val activeAutosellGUIs = HashMap<UUID, AutosellSettingsGUI>()

    fun openPriceEvaluationGUI(player: Player) {
        if (activeEvaluationGUIs.containsKey(player.uniqueId)) player.closeInventory()
        val gui = PriceEvaluationGUI(plugin, player, plugin.getNBTPriceManager())
        activeEvaluationGUIs[player.uniqueId] = gui
        player.openInventory(gui.inventory)
    }

    fun getActivePriceEvaluationGUI(player: Player): PriceEvaluationGUI? = activeEvaluationGUIs[player.uniqueId]

    fun getActivePriceEvaluationGUI(inventory: Inventory): PriceEvaluationGUI? =
        activeEvaluationGUIs.values.firstOrNull { it.inventory == inventory }

    fun openAutosellSettingsGUI(player: Player) {
        if (activeAutosellGUIs.containsKey(player.uniqueId)) player.closeInventory()
        val gui = AutosellSettingsGUI(plugin, player)
        activeAutosellGUIs[player.uniqueId] = gui
        player.openInventory(gui.inventory)
    }

    fun getActiveAutosellSettingsGUI(player: Player): AutosellSettingsGUI? = activeAutosellGUIs[player.uniqueId]

    fun getActiveAutosellSettingsGUI(inventory: Inventory): AutosellSettingsGUI? =
        activeAutosellGUIs.values.firstOrNull { it.inventory == inventory }

    fun removePlayer(player: Player) {
        activeEvaluationGUIs.remove(player.uniqueId)
        activeAutosellGUIs.remove(player.uniqueId)
    }

    fun reload() {
        activeEvaluationGUIs.keys.toSet().forEach { playerId ->
            plugin.server.getPlayer(playerId)?.takeIf { it.openInventory.topInventory.holder is PriceEvaluationGUI }?.closeInventory()
        }
        activeEvaluationGUIs.clear()
        activeAutosellGUIs.keys.toSet().forEach { playerId ->
            plugin.server.getPlayer(playerId)?.takeIf { it.openInventory.topInventory.holder is AutosellSettingsGUI }?.closeInventory()
        }
        activeAutosellGUIs.clear()
    }
}
